src/processor/logic.py

```diff
@@ -0,0 +1,26 @@
+from itertools import chain
+
+from core import db
+
+
+def check_existing_entries(model_column, column_values):
+    """ Get a tuple of values from a model, where entries with those values
+    already exist in the database.
+
+        Args:
+            model_column (object): column to check in the form `Model.Column`
+            column_values (list): list of values to check - these
```

Comment:
I suppose it's reasonable to rely on the common sense of the person passing the `column_values` list into this function, but specifying a set would guarantee unique values.

Reply:
Ah, sorry - I meant values should be for columns where `unique = True` - though it probably is one of those things where common sense should guide the user.